tests/components/tasmota/test_common.py

```diff
@@ -277,34 +334,56 @@ async def help_test_discovery_removal(
 
     async_fire_mqtt_message(hass, f"{DEFAULT_PREFIX}/{config1[CONF_MAC]}/config", data1)
     await hass.async_block_till_done()
+    if sensor_config1:
+        async_fire_mqtt_message(
+            hass,
+            f"{DEFAULT_PREFIX}/{config1[CONF_MAC]}/sensors",
+            json.dumps(sensor_config1),
+        )
+        await hass.async_block_till_done()
 
     # Verify device and entity registry entries are created
     device_entry = device_reg.async_get_device(set(), {("mac", config1[CONF_MAC])})
     assert device_entry is not None
-    entity_entry = entity_reg.async_get(f"{domain}.test")
+    entity_entry = entity_reg.async_get(f"{domain}.{entity_id}")
     assert entity_entry is not None
 
     # Verify state is added
-    state = hass.states.get(f"{domain}.test")
+    state = hass.states.get(f"{domain}.{entity_id}")
     assert state is not None
-    assert state.name == "Test"
+    assert state.name == name
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6319846)